### PR TITLE
Add copious logging to the GCE calls

### PR DESCRIPTION
**Pull request by bowei**
Add copious logging to the GCE calls
```release-note
NONE
```

### CODE REVIEW

**Comment by bowei** [comment]
/assign @nicksardo 

This has patches that are part of a different PR that is currently in the submit queue. I'm going to remove them once the previous PR has merged.

File: pkg/cloudprovider/providers/gce/cloud/gen/main.go [hunk 1]
```diff
@@ -706,33 +716,51 @@ rk := &RateLimitKey{
...
 	}
+
+	if glog.V(4) {
+		glog.V(4).Infof("{{.GCEWrapType}}.List(%v, ..., %v) = [%v items], %v", ctx, fl, len(all), nil)
+	} else if glog.V(5) {
+		var asStr []string
+		for _, o := range all {
+			asStr = append(asStr, fmt.Sprintf("%+v", o))
+		}
+		glog.V(5).Infof("{{.GCEWrapType}}.List(%v, ..., %v) = %v, %v", asStr, nil)
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/cloud/gen/main.go]: Missing values
`List(%v, ..., %v) = %v, %v", ctx, fl, asStr, nil)`

File: pkg/cloudprovider/providers/gce/cloud/gen/main.go [hunk 2]
```diff
@@ -823,13 +870,24 @@ func (g *{{.GCEWrapType}}) AggregatedList(ctx context.Context, fl *filter.F) (ma
...
 		return nil, err
 	}
+	if glog.V(4) {
+		glog.V(4).Infof("{{.GCEWrapType}}.AggregatedList(%v, %v) = [%v items], %v", ctx, fl, len(all), nil)
+	} else if glog.V(5) {
+		var asStr []string
+		for _, o := range all {
+			asStr = append(asStr, fmt.Sprintf("%+v", o))
+		}
+		glog.V(5).Infof("{{.GCEWrapType}}.AggregatedList(%v, %v) = %v, %v", asStr, nil)
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/cloud/gen/main.go]: Same issue

File: pkg/cloudprovider/providers/gce/cloud/gen/main.go [hunk 3]
```diff
@@ -874,21 +938,37 @@ func (g *{{.GCEWrapType}}) {{.FcnArgs}} {
...
+	return v, err
 {{- else if .IsPaged}}
 	var all []*{{.Version}}.{{.ItemType}}
 	f := func(l *{{.Version}}.{{.ReturnType}}) error {
+		glog.V(5).Infof("{{.GCEWrapType}}.{{.Name}}(%v, %v, ...): page %+v", ctx, key, l)
 		all = append(all, l.Items...)
 		return nil
 	}
 	if err := call.Pages(ctx, f); err != nil {
+		glog.V(4).Infof("{{.GCEWrapType}}.{{.Name}}(%v, %v, ...) = %v, %v", nil, err)
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/cloud/gen/main.go]: Same problem

File: pkg/cloudprovider/providers/gce/cloud/gen/main.go [hunk 4]
```diff
@@ -874,21 +938,37 @@ func (g *{{.GCEWrapType}}) {{.FcnArgs}} {
...
 		return nil, err
 	}
+	if glog.V(4) {
+		glog.V(4).Infof("{{.GCEWrapType}}.{{.Name}}(%v, %v, ...) = [%v items], %v", len(all), nil)
+	} else if glog.V(5) {
+		var asStr []string
+		for _, o := range all {
+			asStr = append(asStr, fmt.Sprintf("%+v", o))
+		}
+		glog.V(5).Infof("{{.GCEWrapType}}.{{.Name}}(%v, %v, ...) = %v, %v", asStr, nil)
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/cloud/gen/main.go]: Same problem

File: pkg/cloudprovider/providers/gce/cloud/gen/main.go [hunk 5]
```diff
@@ -874,21 +938,37 @@ func (g *{{.GCEWrapType}}) {{.FcnArgs}} {
...
+		glog.V(5).Infof("{{.GCEWrapType}}.{{.Name}}(%v, %v, ...): page %+v", ctx, key, l)
 		all = append(all, l.Items...)
 		return nil
 	}
 	if err := call.Pages(ctx, f); err != nil {
+		glog.V(4).Infof("{{.GCEWrapType}}.{{.Name}}(%v, %v, ...) = %v, %v", nil, err)
 		return nil, err
 	}
+	if glog.V(4) {
+		glog.V(4).Infof("{{.GCEWrapType}}.{{.Name}}(%v, %v, ...) = [%v items], %v", len(all), nil)
```
Comment (nicksardo) [inline pkg/cloudprovider/providers/gce/cloud/gen/main.go]: Same problem

**Pull request merged**